src/NLog/LayoutRenderers/LayoutRenderer.cs

```diff
@@ -49,12 +49,18 @@ public abstract class LayoutRenderer : ISupportsInitialize, IRenderable, IDispos
         private const int MaxInitialRenderBufferLength = 16384;
         private int _maxRenderedLength;
         private bool _isInitialized;
+        private IValueFormatter _valueFormatter;
 
         /// <summary>
         /// Gets the logging configuration this target is part of.
         /// </summary>
         protected LoggingConfiguration LoggingConfiguration { get; private set; }
 
+        /// <summary>
+        /// Value formatter
+        /// </summary>
+        protected IValueFormatter ValueFormatter => _valueFormatter ?? (_valueFormatter = Resolve<IValueFormatter>());
```

Comment:
Missing fallback to MessageTemplates.ValueFormatter.Instance

Reply:
It's in the container, and you can register `null`